jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Connection.java

```diff
@@ -421,16 +421,44 @@ public void onConnectionFailure(int error, String reason)
         }
     }
 
-    private class NetworkBuffer extends RetainableByteBuffer implements Callback
+    private class NetworkBuffer implements Callback
```

Comment:
It would easily take as much code to allow `RetainableByteBuffer` to be sub-classed in such a way that an instance can delegate its lifecycle management to another instance.

I preferred to keep `RetainableByteBuffer` as focused as possible at the expense of 5 one-liner methods in this class.